### DISCUSSION TITLE
NANO ELS H4 ON GRIZZLY G0768 LATHE WITH BELT CROSSLIDE

### DISCUSSION THREAD

**Discussion by RGUERI1** [kachurovskiy/nanoels#129] [Show and tell]
I can't thank Maxim enough for all of the speedy responses and help on this build. I've wanted to install an ELS since getting my G0768 lathe but couldn't find many examples of this model. I took the plunge and decided to figure it out. After a few spools of 3D printing filament, I think I have a stable working system. Just a little info on the pictures and why I had to locate things the way I did:

**_Spindle Encoder-_** due to the dual speed pulleys on this unit, I ended up having to relocate the encoder numerous times. I ended up going with a 2:1 ratio with the empty low speed pulley on my spindle. I ordered the same 5M375 V-belt that drives the spindle and printed a pulley 1/2 the diameter. It took a little tweaking but a 40.3mm O.D. ended up working.

_**Leadscrew Drive-**_  this lathe has a reversable leadscrew gear box under the speed control cavity so I couldn't find any way to mount the stepper motor without extensive modifications. My goal was keep all ELS components within the original footprint of the machine. However, I ended up having to mount it on the right side. It doesn't seem to get in the way, but I'd like to eventually mount it under the ways somehow and add a belt and pulley, but its working for now. I had to modify the 14mm hex nut on the end of the leadscrew with a set screw, just to ensure it doesn't back off. I'm using a 3D printed adapter from 14mm hex to the 5mm stepper shaft and it seems to be working. I did however break a couple of them during tuning, I've increased the thickness and infill and its been doing good since. I will eventually re-make everything out of aluminum once I'm satisfied with the overall operation. The stepper bracket also had to be beefed up and some gussets added for rigidity. 

_**Cross Slide Drive-**_ I originally went with the gear driven version but during my build, the belt drive system came out, so I went with that for the final. I did some modifications to the one on Thingiverse because I wanted to retain my original handwheel and not use a printed one. The solution was to make it a press-fit around the original indicator shaft on the handwheel. This incorporates the pulley on the original handwheel and has been working good so far. I also had to add an offset on the face that contacts the front of my cross slide, this is where the horizontal and vertical plate bolt up and don't align flush on my machine. Another modification I made to the design, was to add thrust bearings on both sides of the cross slide leadscrew securement to the new housing. This definitely makes a smoother and tighter feel when running it manually. One last mod was having to mill out a shallow channel on the underside of the cross slide to accommodate the height of the new housing. 

Overall it was a lot slower process than I expected but worth it in the long run. It also gave me an excuse/opportunity to adjust and clean my machine, as it was long overdue. I do all my 3D design in Sketchup, but if someone needs anything in another format, just let me know and I can get it for you. I highly recommend the thrust bearings they are a cheap fix and well worth the time and effort. 
Thanks again to Maxim for all his hard work and effort, much appreciated.

Now I can confidently cut that one set of threads that I need every couple of years, with confidence and no change gears!
![OVERALL COMPLETE](https://github.com/kachurovskiy/nanoels/assets/362667/79d6048b-b063-45d6-9e3c-ed19d4c4a204)
![NANO H4 MOUNT](https://github.com/kachurovskiy/nanoels/assets/362667/08604e94-23ed-4bfd-af95-e034bdf03ae5)
![IMG_1378](https://github.com/kachurovskiy/nanoels/assets/362667/62257216-726c-465d-9c7b-c180b7bfb305)
![IMG_1381](https://github.com/kachurovskiy/nanoels/assets/362667/6b8214ea-e27b-41da-a8f8-3654b20b2f9f)
![IMG_1379](https://github.com/kachurovskiy/nanoels/assets/362667/ba8340c5-5424-4f7c-9dea-0ce23a5d7485)
![IMG_1675](https://github.com/kachurovskiy/nanoels/assets/362667/2d962c81-abf5-416f-9e02-3ff3c8614aa4)
![IMG_1677](https://github.com/kachurovskiy/nanoels/assets/362667/8d1d42fc-45c0-4205-97b1-712b68783e3d)
![IMG_1678](https://github.com/kachurovskiy/nanoels/assets/362667/c5e31807-a7d4-43b9-b04c-dd2c3f79ba99)
![IMG_1682](https://github.com/kachurovskiy/nanoels/assets/362667/c835c896-6e95-445b-8d45-cc4d1000113d)
![CROSS SLIDE MILL FOR BEARING](https://github.com/kachurovskiy/nanoels/assets/362667/c8275554-81ad-407e-89f9-d2106635d979)
![CROSS SLIDE RIGHT VIEW](https://github.com/kachurovskiy/nanoels/assets/362667/b6192d44-3eed-47c5-a4e7-385ac30650d2)
![IMG_1393](https://github.com/kachurovskiy/nanoels/assets/362667/24c0e8a0-0d5c-42d1-bed6-e0e6e047a3fd)
![OFFSET FOR FLUSH PLATE](https://github.com/kachurovskiy/nanoels/assets/362667/9b94f54d-8327-4936-aca4-dcaa5acfa17a)
[CROSS SLIDE SKETCHUP FILES.zip](https://github.com/kachurovskiy/nanoels/files/12398257/CROSS.SLIDE.SKETCHUP.FILES.zip)
[LEADSCREW BRACKET AND HEX ADAPTER SKETCHUP FILES.zip](https://github.com/kachurovskiy/nanoels/files/12398262/LEADSCREW.BRACKET.AND.HEX.ADAPTER.SKETCHUP.FILES.zip)
[SPINDLE ENCODER 40.3MM SKETCHUP FILES.zip](https://github.com/kachurovskiy/nanoels/files/12398274/SPINDLE.ENCODER.40.3MM.SKETCHUP.FILES.zip)

**Comment by kachurovskiy** [comment]
Hi @RGUERI1, what an awesome build, thank you so much for sharing! Very well done.

**Comment by RGUERI1** [comment]
Thanks for the feedback!

**Comment by RGUERI1** [comment]
Maxim,

Just wanted to give an update on my setup, since I've had a few chances to use it since completing the system. 

- I've added two encoder wheels for the Z and X axis. I thought this would be a good improvement since it's so close to cross-slide motor enclosure, at the Z handwheel. However, I'm not comfortable yet with the accuracy/slop in the output when using the wheels. I have to turn a few degrees before it recognizes the intended direction. I also have to move it rather quickly in that direction to get it to recognize the signal. Is there any tuning or code adjustments that would help make this a smoother transition? I've played with the `PULSE_HALF_BACKLASH` setting in the code for it and that does seem to help, I think I currently have it set to 3. I was considering trying to incorporate or swap the encoder wheels out with a joystick or a directional switch. Any thoughts on doing that or has anyone else tried something similar? 
- I have experienced some loss of steps on the leadscrew stepper, when trying to do deeper cuts than I should have, being that I have the CLOSED LOOP stepper motor, should I have wired something else in order to notify the NANO that a step was missed/skipped? 
- I'm growing accustomed to the leadscrew stepper hanging off the right side, it's not in the way as much as I thought. However, I have broken a couple of the 3D-printed 14-mm hexnut to stepper shaft adapters and they do introduce some extra backlash, so I'm working on a better design for that part. I'm thinking of drilling and installing a roll pin.

**Reply by kachurovskiy** [reply]
@RGUERI1 thank you for the feedback!

I've seen similar encoder problems as well. They could be caused by insufficient current being available for the encoders through the A1 and A2 5V pin. Consider connecting encoder GND and 5V pins to a power source directly and connecting just the 2 signal lines to the A1 and A2 via a shielded cable. Hope it helps. Then you'd hopefully be able to reduce `PULSE_HALF_BACKLASH` to 1 or 0 and get rid of the direction change delay. Please also post some pictures of your current setup and your encoder models to make sure we're on the same page.

I've hit the error state on my stepper drivers too. They just stop reacting to commands until ENA is toggled. H4 doesn't have a built-in way to react to external ESTOP signals but you could add such logic e.g. by connecting your driver ALERT output (assuming it's 5V) to one of the available pins on A1/A2 and adding code to listen for that signal and calling `setEmergencyStop(...)`. It might be easier/better to just cut the power to your entire system when that happens though.

Those hex adapters break do to layer orientation. Consider printing them with layer lines going alongside the leadscrew and perhaps increase the external dimension? Good luck!

**Reply by kachurovskiy** [reply]
If you're using [MPG-style](https://m.media-amazon.com/images/I/51QcEwm0xSL._AC_SX679_.jpg) encoders, consider trying optical rotary encoders instead - they've performed much more predictable for me so far even when powered from the A1/A2 terminals directly.

**Comment by RGUERI1** [comment]
Ok, I will give all of those suggestions a try. Thanks.

**Reply by RGUERI1** [reply]
I did have problems with one of the encoders, both purchased off Amazon at the same time, but appear to be different brands. I didn't hook it up to the scope to see what kind of signals I was getting but it was very inconsistent. I swapped it around to the other axis and the problem followed, so I knew it was in the encoder. I ordered a third and it worked identical to the other good one.

**Comment by Festivejelly** [comment]
Nice solution. I think my warco WM180 looks similar to the internals you have. What size belt did you use to connect to the rotary encoder? This looks like the least intrusive to me!
